File: my-app/src/services/clientService.js
// src/services/clientService.js
import { api } from './api';

// 클라이언트 정보 조회
export const getClientInfo = async (clientId) => {
  try {
    const response = await api.get(`/api/clients/${clientId}`);
    return response;
  } catch (error) {
    console.error('Error fetching client info:', error);
    
    // 네트워크 오류시 더미 데이터 반환
    if (error.isNetworkError) {
      console.log('네트워크 오류, 더미 클라이언트 데이터 사용');
      return {
        age: "30대",
        occupation: "회사원",
        goal: "직장 스트레스 관리",
        note: "불면증 호소, 업무 과부하"
      };
    }
    
    throw error;
  }
};

// 클라이언트 정보 업데이트
export const updateClientInfo = async (clientId, clientData) => {
  try {
    const response = await api.put(`/api/clients/${clientId}`, clientData);
    return response;
  } catch (error) {
    console.error('Error updating client info:', error);
    
    // 네트워크 오류시 성공한 것처럼 처리 (UI 업데이트 유지)
    if (error.isNetworkError) {
      console.log('네트워크 오류, 로컬 UI만 업데이트됨');
      return clientData; // 입력한 데이터 그대로 반환
    }
    
    throw error;
  }
};

File: counsleBE/src/main/java/com/example/sociallogin/service/ClientService.java
package com.example.sociallogin.service;

import com.example.sociallogin.domain.Client;
import com.example.sociallogin.dto.ClientDTO;
import com.example.sociallogin.repository.ClientRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import java.util.Date;
import java.util.Optional;

@Slf4j
@Service
@RequiredArgsConstructor
public class ClientService {

    private final ClientRepository clientRepository;

    public ClientDTO getClientInfo(String clientId) {
        log.info("Retrieving client info for clientId: {}", clientId);

        Optional<Client> clientOpt = clientRepository.findById(clientId);

        if (clientOpt.isEmpty()) {
            log.warn("Client not found with ID: {}", clientId);
            return new ClientDTO(); // 빈 DTO 반환
        }

        Client client = clientOpt.get();
        return ClientDTO.fromEntity(client);
    }

    public ClientDTO updateClientInfo(ClientDTO clientDTO) {
        log.info("Updating client info for clientId: {}", clientDTO.getId());

        // 기존 클라이언트 조회
        Optional<Client> existingClientOpt = clientRepository.findById(clientDTO.getId());

        Client clientToSave;

        if (existingClientOpt.isPresent()) {
            // 기존 클라이언트 업데이트
            Client existingClient = existingClientOpt.get();

            // 필드 업데이트
            existingClient.setAge(clientDTO.getAge());
            existingClient.setOccupation(clientDTO.getOccupation());
            existingClient.setGoal(clientDTO.getGoal());
            existingClient.setNote(clientDTO.getNote());
            existingClient.setUpdatedAt(new Date());

            clientToSave = existingClient;
        } else {
            // 새로운 클라이언트 생성
            clientToSave = Client.builder()
                    .id(clientDTO.getId())
                    .age(clientDTO.getAge())
                    .occupation(clientDTO.getOccupation())
                    .goal(clientDTO.getGoal())
                    .note(clientDTO.getNote())
                    .createdAt(new Date())
                    .updatedAt(new Date())
                    .build();
        }

        // 저장 후 결과 반환
        Client savedClient = clientRepository.save(clientToSave);
        return ClientDTO.fromEntity(savedClient);
    }
}